File: src/test/java/io/veridia/jsonlogic/AllExpressionTests.java
package io.veridia.jsonlogic;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class AllExpressionTests {
  private static final JsonLogic jsonLogic = new JsonLogic();

  @Test
  public void testEmptyArray() throws JsonProcessingException {
    assertEquals(false, jsonLogic.apply("{\"all\": [[], {\">\": [{\"var\": \"item\"}, 0]}]}", null));
  }

  @Test
  public void testAll() throws JsonProcessingException {
    assertEquals(true, jsonLogic.apply("{\"all\": [[1, 2, 3], {\">\": [{\"var\": \"item\"}, 0]}]}", null));
    assertEquals(false, jsonLogic.apply("{\"all\": [[1, 2, 3], {\">\": [{\"var\": \"item\"}, 1]}]}", null));
  }
}

File: src/test/scala/io/veridia/jsonlogic/AllExpressionTests.scala
package io.veridia.jsonlogic

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class AllExpressionTests extends AnyFunSuite with Matchers:
  private val jsonLogic = new JsonLogic()

  test("empty array is false") {
    jsonLogic.apply("{\"all\": [[], {\">\": [{\"var\": \"item\"}, 0]}]}", null) shouldEqual false
  }

  test("all") {
    jsonLogic.apply("{\"all\": [[1, 2, 3], {\">\": [{\"var\": \"item\"}, 0]}]}", null) shouldEqual true
    jsonLogic.apply("{\"all\": [[1, 2, 3], {\">\": [{\"var\": \"item\"}, 1]}]}", null) shouldEqual false
  }
